File: 2089. Find Target Indices After Sorting Array/Optimized_Solution.cpp
class Solution {
public:
    vector<int> targetIndices(vector<int>& nums, int target) {
        int less_count = 0;
        int count = 0;
        
        for (int num : nums) {
            if (num < target) {
                less_count++;
            } else if (num == target) {
                count++;
            }
        }
        
        vector<int> result;
        for (int i = less_count; i < less_count + count; i++) {
            result.push_back(i);
        }
        
        return result;
    }
};

File: 2089. Find Target Indices After Sorting Array/Optimized_Solution.py
class Solution:
    def targetIndices(self, nums: List[int], target: int) -> List[int]:
        less_count = 0
        count = 0
        
        for num in nums:
            if num < target:
                less_count += 1
            elif num == target:
                count += 1
        
        return list(range(less_count, less_count + count)) 
#Time complexity - O(n)
